Products/DataCollector/zing/fact.py

```diff
@@ -78,7 +78,14 @@ def _tweak_data(self, data_in):
         data_out = {}
         for k, v in data_in.iteritems():
             if isinstance(v, list) or isinstance(v, tuple) or isinstance(v, set):
-                data_out[k] = sorted(v)
+                # whatever comes in the list, tuple etc, needs to be scalar
+                # if not, cast it to string
+                values = []
+                for x in v:
+                    if not isinstance(x, (str, int, float, bool)):
+                        x = str(x)
+                    values.append(x)
```

Comment:
We lose the "grouping" aspect of the list-of-list or the list-of-dict in this case. That's okay for now, but should think about how we can propagate that info...